src/main/java/core/basesyntax/impl/StorageImpl.java

```
@@ -1,19 +1,48 @@
 package core.basesyntax.impl;
 
 import core.basesyntax.Storage;
+import java.util.Objects;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_CAPACITY = 10;
+    private final K[] keys;
+    private final V[] values;
+    private int currentSize;
+
+    @SuppressWarnings("unchecked")
+    public StorageImpl() {
+        this.keys = (K[]) new Object[MAX_CAPACITY];
+        this.values = (V[]) new Object[MAX_CAPACITY];
+        currentSize = 0;
+    }
+
     @Override
     public void put(K key, V value) {
+        for (int i = 0; i < currentSize; i++) {
+            if (Objects.equals(keys[i], key)) {
+                values[i] = value;
+                return;
+            }
+        }
+        if (currentSize < MAX_CAPACITY) {
+            keys[currentSize] = key;
+            values[currentSize] = value;
+            currentSize++;
+        }
```

The `put` method does not handle the case where the storage reaches its maximum capacity effectively. It silently fails to add new elements once the capacity is reached. Consider throwing an exception or providing feedback to the caller when the storage is full.